x-pack/plugins/uptime/public/components/monitor/ml/ml_flyout_container.tsx

```diff
@@ -120,14 +119,14 @@ export const MachineLearningFlyout: React.FC<Props> = ({ onClose }) => {
           hasMLJob.awaitingNodeAssignment,
           core.services.theme?.theme$
         );
-        const loadMLJob = (jobId: string) =>
-          dispatch(getExistingMLJobAction.get({ monitorId: monitorId as string }));
-
-        loadMLJob(ML_JOB_ID);
-
+        dispatch(getExistingMLJobAction.get({ monitorId: monitorId as string }));
         refreshApp();
-        dispatch(setAlertFlyoutType(CLIENT_ALERT_TYPES.DURATION_ANOMALY));
-        dispatch(setAlertFlyoutVisible(true));
+
+        const hasUptimeWrite = core.services.application?.capabilities.uptime?.save ?? false;
```

Comment:
i am not sure why we are checking uptime.save capability, shouldn't we check if user has alerting save capability?

`alerting:save`

Reply:
Not really, that's not the capability used to save alerts. I did discuss this with Justin and he mentioned that we actually rely on save, not on `alerting:save`, which is, by the way, always false, even when users have all write permissions.